Look at [mapping_Dvir_DiNV_combined.md](https://github.com/meschedl/DiNV-Dv1-Genome-Integration/blob/main/Mapping/mapping_Dvir_DiNV_combined.md) for how I mapped reads to a concatinated DiNV + D. vrirlis genome 

Look at [chimeric_alignments.md](https://github.com/meschedl/DiNV-Dv1-Genome-Integration/blob/main/Mapping/chimeric_alignments.md) for how I separated out chimeric alignments from the mapping (mapped to both vrilis and DiNV), and how I tried to figure out if they were real (this file is long and messy)

Look at [check_chimeric_alignments.md](https://github.com/meschedl/DiNV-Dv1-Genome-Integration/blob/main/Mapping/check_chimeric_alignments.md) for how I confrimed that none of the chimeric alignments were true chimeric alignments by looking at how they align to the read 

[chimeric_alignments_stats.csv](https://github.com/meschedl/DiNV-Dv1-Genome-Integration/blob/main/Mapping/chimeric_alignments_stats.csv) is by eye classifications of whether I thought an alignment could be chimeric or not (none are though)

[all_chimeric_positions_positions.csv](https://github.com/meschedl/DiNV-Dv1-Genome-Integration/blob/main/Mapping/all_chimeric_positions_positions.csv) is info on all alignments that came out as chimeric (they're not though)

[Primers_for_chimeric_alignments.md](https://github.com/meschedl/DiNV-Dv1-Genome-Integration/blob/main/Mapping/Primers_for_chimeric_alignments.md) is where I tried to generate primers for PCRing around potential chimeric junctions, and where I begain to think the junctions didn't exist
